app/src/main/java/com/cmput301f20t21/bookfriends/services/AuthService.java

```diff
@@ -43,4 +49,21 @@ public User getCurrentUser() {
     public void signOut() {
         mAuth.signOut();
     }
+
+    public void updateEmail(String username, String TAG){
+        FirebaseUser user = mAuth.getCurrentUser();
+        if (user != null) {
+            user.updateEmail(username)
+                    .addOnCompleteListener(new OnCompleteListener<Void>() {
```

Comment:
the OnCompleteHandler should exist in ViewModel instead of service, because currently there's no way for the fragment to know if the update succeed or failed(Log.d is only for developer debugging purposes and user cannot see that) 

the service should just do `return user.updateEmail(...)`, the ViewModel should have an instance of the service and call `authService.updateEmail(...)` then add handlers that can be directly pass back to the fragment.

Also I think the parameter should be `email` instead of `username` ? I don't see the email that is being updated here

If you are unsure how to proceed we can discuss during tonight's meeting and show an example from another fragment